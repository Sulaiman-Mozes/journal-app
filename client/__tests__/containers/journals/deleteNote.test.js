import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import DeleteNotePage, { DeleteNote } from '../../../containers/journals/deleteNote';

describe('JournalPage Container', () => {
  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  const initialState = {};
  const store = mockStore(initialState);

  const props = {
    deleteFunc: jest.fn(),
    id: '6347ewytug7854uieajksfhv9ioq34',
  };

  it('should render correctly', () => {
    const deleteContainer = shallow(
      <Provider store={store}>
        <DeleteNotePage {...props} />
      </Provider>,
    );
    expect(deleteContainer).toMatchSnapshot();
  });

  it('should have the following methods', () => {
    const wrapper = shallow(<DeleteNote {...props} />);
    expect(wrapper.instance().handleDelete());
  });
});
