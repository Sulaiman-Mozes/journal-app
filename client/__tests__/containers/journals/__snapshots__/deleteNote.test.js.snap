// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JournalPage Container should render correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeState": Object {},
    }
  }
>
  <Connect(DeleteNote)
    deleteFunc={[MockFunction]}
    id="6347ewytug7854uieajksfhv9ioq34"
  />
</ContextProvider>
`;
